sockeye/training.py

```diff
@@ -223,16 +238,25 @@ def _fit(self,
         """
         metric_train = self._create_eval_metric(metrics)
         metric_val = self._create_eval_metric(metrics)
-        num_not_improved = 0
         tic = time.time()
-        epoch = 0
-        checkpoint = 0
-        updates = 0
-        samples = 0
+
+        training_state_dir = os.path.join(output_folder, C.TRAINING_STATE_DIRNAME)
+        if os.path.exists(training_state_dir):
+            training_state = self.load_checkpoint(training_state_dir, train_iter)
+        else:
+            training_state = _TrainingState(
```

Comment:
why set the defaults explicitly if you have them in your minimal TrainingState class?
If we make them explicit here we can just as well use typing.NamedTuple and have type annotations as in AttentionState/DecoderState